crates/oxc_traverse/src/walk.rs

```diff
@@ -2228,7 +2228,7 @@ pub(crate) unsafe fn walk_function<'a, Tr: Traverse<'a>>(
 ) {
     traverser.enter_function(&mut *node, ctx);
     ctx.push_stack(Ancestor::FunctionId(ancestor::FunctionWithoutId(node)));
-    let has_scope = matches!(ctx.ancestor(2).unwrap(), Ancestor::MethodDefinitionValue(_));
+    let has_scope = !matches!(ctx.ancestor(2).unwrap(), Ancestor::MethodDefinitionValue(_));
```

Comment:
I found a bug about the `ScopeFlags` of the constructor method is incorrect. The `JSXOpeningElement` is inside the constructor method, But I get the scope is `ScopeFlags::Function` and the parent scope is `ScopeFlags::Constructor`. It seems that both construct and function have pushed a scope flag.

Reply:
Ah. I will investigate. This shouldn't happen.